packages/@react-spectrum/list/docs/ListView.mdx

```diff
@@ -0,0 +1,941 @@
+        break;
+      case 'after':
+        list.moveAfter(target.key, keys);
+        break;
+      case 'on': {
+        let targetFolder = list.getItem(target.key);
+        let draggedItems = keys.map((key) => list.getItem(key));
+        list.update(target.key, {...targetFolder, childNodes: [...targetFolder.childNodes, ...draggedItems]});
+        list.remove(...keys);
+        break;
+      }
+    }
+  };
+
+  let dragHooks = useDragHooks({
+    getItems(keys) {
+      return [...keys].map(key => {
+        let item = list.getItem(key);
+        // Setup the info for each item that will be propagated upon drag
+        return {
+          [`${dragType}-${item.type}`]: JSON.stringify(item)
+        };
+      });
+    }
+  });
+
+  let dropHooks = useDropHooks({
+    async onDrop(e) {
+      let keysToMove = [];
+      // Only add the dropped item to the list of keys to move
+      // if the dropped item has the expected types
+      for (let item of e.items) {
+        for (let acceptedType of acceptedDragTypes) {
+          if (item.kind === 'text' && item.types.has(acceptedType)) {
+            let {id} = JSON.parse(await item.getText(acceptedType));
+            keysToMove.push(id);
+          }
+        }
+      }
+      onMove(keysToMove, e.target);
+    },
+    getDropOperation(target, types) {
+      let typesSet = types.types ? types.types : types;
+      let draggedTypes = [...typesSet.values()];
+      // Cancel the drop operations if any of drag items aren't of the acceptedTypes, if the drop target is root,
+      // or if attempting to drop on a non-folder item
+      if (
+        !draggedTypes.every(type => acceptedDragTypes.includes(type)) ||
+        target.type === 'root' ||
+        (target.type === 'item' && target.dropPosition === 'on' && !list.getItem(target.key).childNodes)
+      ) {
+        return 'cancel';
+      }
+
+      return 'move';
+    }
+  });
+
+  return (
+    <ListView
+      aria-label="Reorderable ListView"
+      selectionMode="multiple"
+      width="size-3600"
+      height="size-3600"
+      items={list.items}
+      dragHooks={dragHooks}
+      dropHooks={dropHooks}>
+      {item => (
+        <Item textValue={item.name} hasChildItems={item.type === 'folder'}>
+          <Text>{item.name}</Text>
+          {item.type === 'folder' &&
+            <>
+              <Folder />
+              <Text slot="description">{`contains ${item.childNodes.length} dropped item(s)`}</Text>
+            </>
+          }
+        </Item>
+      )}
+    </ListView>
+  );
+}
+```
+</details>
+
+```tsx example
+<ReorderableList />
+```
+
+The example below demonstrates how to create a pair of ListViews that supports dragging and dropping any items between each list, but disables
+the ability to drop into a folder and onto the root of the list.
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
```

Comment:
this example is a bit strange, you can empty the list, but once you do, you can't drag anything into it because root drop is not allowed. Until that point you can drag back and forth all you like.